### ISSUE TITLE
Html2Text.convert() blows out on non-string

### ISSUE THREAD

**Issue by vpithart** [soundasleep/html2text_ruby#2]
## How to reproduce?

```
[5] pry(main)> Html2Text.convert('<b>ab</b>cd')
=> "abcd"
```
```
[6] pry(main)> Html2Text.convert(nil)
NoMethodError: undefined method `gsub' for nil:NilClass
from /Users/vpithart/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/html2text-0.2.0/lib/html2text.rb:22:in `replace_entities'
```
```
[9] pry(main)> Html2Text.convert(0)
NoMethodError: undefined method `gsub' for 0:Fixnum
from /Users/vpithart/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/html2text-0.2.0/lib/html2text.rb:22:in `replace_entities'
```

## Desired behaviour
The `convert()` should internally call `.to_s` on whatever the input is.

**Comment by vpithart** [comment]
Hey @soundasleep, hereby I provide a patch that fixes this issue. Including tests. Enjoy.

```
From 410f34f0951a17da842004d4ed433821aa96b3ab Mon Sep 17 00:00:00 2001
From: Vojtech Pithart <vojtech.pithart@wealthbar.com>
Date: Thu, 21 Sep 2017 09:20:45 +0200
Subject: [PATCH] Fixed convert() for non-string input

---
 lib/html2text.rb         |  1 +
 lib/html2text/version.rb |  2 +-
 spec/html2text_spec.rb   | 21 +++++++++++++++++++++
 3 files changed, 23 insertions(+), 1 deletion(-)

diff --git a/lib/html2text.rb b/lib/html2text.rb
index 228138c..f5126d0 100644
--- a/lib/html2text.rb
+++ b/lib/html2text.rb
@@ -8,6 +8,7 @@ class Html2Text
   end

   def self.convert(html)
+    html = html.to_s
     html = fix_newlines(replace_entities(html))
     doc = Nokogiri::HTML(html)

diff --git a/lib/html2text/version.rb b/lib/html2text/version.rb
index b45e0ab..a072f4f 100644
--- a/lib/html2text/version.rb
+++ b/lib/html2text/version.rb
@@ -1,3 +1,3 @@
 class Html2Text
-  VERSION = "0.2.0"
+  VERSION = "0.2.1"
 end
diff --git a/spec/html2text_spec.rb b/spec/html2text_spec.rb
index a4d5328..dd95bd6 100644
--- a/spec/html2text_spec.rb
+++ b/spec/html2text_spec.rb
@@ -19,6 +19,27 @@ describe Html2Text do
         expect(text).to eq("hello world")
       end
     end
+
+    context "input value is non-string" do
+      let(:html) { nil }
+      it '(nil)' do
+        expect(text).to eq("")
+      end
+    end
+
+    context "input value is non-string" do
+      let(:html) { 1234 }
+      it "(number)" do
+        expect(text).to eq("1234")
+      end
+    end
+
+    context "input value is non-string" do
+      let(:html) { 1234.5600 }
+      it "(float number)" do
+        expect(text).to eq("1234.56")
+      end
+    end
   end

   describe "#remove_leading_and_trailing_whitespace" do
--
2.12.2
```

**Comment by soundasleep** [comment]
This is great, thank you! Can you put that into a pull request? <3

**Comment by vpithart** [comment]
I tried to do so already, but gitlab didn't allow me to push to this project:

    git push --set-upstream origin resiliency-against-non-string
    fatal: unable to access 'https://github.com/soundasleep/html2text_ruby.git/': The requested URL returned error: 403

Perhaps you need to add me to contributors?

**Comment by soundasleep** [comment]
Hi! To make a PR on GitHub you need to Fork the project into your own repository, then push the changes there - and then create a Pull Request using your fork as the source. https://help.github.com/articles/creating-a-pull-request-from-a-fork/

By contributing a PR you'll also be credited as a contributor to the project. :)

**Comment by soundasleep** [comment]
Resolved with PR #3
